__tests__/fetchers.js

```diff
@@ -206,3 +206,34 @@ test('TarballFetcher.fetch properly stores tarball of scoped package in offline
   const exists = await fs.exists(path.join(offlineMirrorDir, '@exponent-configurator-1.0.2.tgz'));
   expect(exists).toBe(true);
 });
+
+test.only('TarballFetcher.fetch retry on 500', async () => {
+  // jest.useRealTimers();
+  const dir = await mkdir('tarball-fetcher');
+  jest.mock('request', factory => options => {
+    return {
+      on: (name, cb) => {
+        if (name === 'response') {
```

Comment:
Thanks, @torifat, if you are too busy right now it would be better to merge the fix and add the test to the backlog.